File: Clases/Unidad02_POO/C#/Publicaciones_v04/Publicaciones_v04/Publicacion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones_v04
{
    internal abstract class Publicacion
    {
        // Atributos
        public static int orden_presentacion = 1;
        protected string nombre;
        protected string[] autores;
        protected int cantidad_autores;
        protected int año;
        // Metodos
        // Constructor
        public Publicacion(string nombre, string[] autores, int cantidad_autores, int año)
        {
            this.nombre = nombre;
            this.autores = autores;
            this.cantidad_autores = cantidad_autores;
            this.año = año;
        }
        public Publicacion(string nombre, string autor, int año)
        {
            this.nombre = nombre;
            this.autores = new string[1];
            this.autores[0] = autor;
            this.cantidad_autores = 1;
            this.año = año;
        }
        // Impresion
        public abstract void referenciar();
    }
}

File: Clases/Unidad03_Librerias_Componentes/Java/Publicaciones_v03_b/src/main/java/Publicaciones_v03_b/publicaciones/Publicacion.java
package Publicaciones_v03_b.publicaciones;

import java.util.ArrayList;

public abstract class Publicacion{
// Atributos
    public static Integer orden_presentacion = 1;
    protected String nombre;
    protected ArrayList<Autor> autores;
    protected Integer cantidad_autores;
    protected Integer año;
// Metodos
    // Constructor
    public Publicacion(String nombre, String autores[], Integer año){
        this.nombre = nombre;
        this.autores = new ArrayList<Autor>();
        for(String nombres:autores){
            Autor nombreAutor = new Autor(nombres);
            this.autores.add(nombreAutor);
        }
        this.cantidad_autores = this.autores.size();
        this.año = año;
    }
    public Publicacion(String nombre, String autor, Integer año){
        this.nombre = nombre;
        this.autores = new ArrayList<Autor>();
        Autor nombreAutor = new Autor(autor);
        this.autores.add(nombreAutor);
        this.cantidad_autores = 1;
        this.año = año;
    }
    // Impresion
    public abstract void referenciar();
}
